docs/src/main/asciidoc/extension-maturity-model.adoc

```diff
@@ -0,0 +1,93 @@
+
+[id="extension-maturity-model"]
+= A maturity model for Quarkus extensions
+include::_attributes.adoc[]
+:diataxis-type: concept
+:categories: writing-extensions
+:topics: extensions
+:summary: Quarkus extensions can do a lot, or a little. This guide explains some of the capabilities extension authors may wish to include.
+////
+The document header ends at the first blank line. Do not remove the blank line between the header and the abstract summary.
+////
+
+What makes a good Quarkus extension? What capabilities is a Quarkus extension expected to provide?
+
+* It actually works
+** Works in JVM mode
+** Works in dev mode
+** Works in native
+* Dev service (if there is an external service dependency)
```

Comment:
shouldn't this not be under developer joy?

Reply:
I wanted to pull it out to the top level, because it's so important. 

For consistency I could also pull config out to the top level, and leave developer joy to be the more amorphous 'take advantage of quarkus magic to revolutionise your programming model.'